### ISSUE TITLE
getByText triggers a memory leak warning with jest's --detectLeaks flag

### ISSUE THREAD

**Issue by bitttttten** [testing-library/react-testing-library#389]
```
react-testing-library@6.0.3 
react@16.8.6 
node v10.15.2
npm 6.4.1
```

### Relevant code or config:

```js
/** @jsx jsx */
import { jsx } from '@emotion/core'
import { render } from '@testing-library/react'
import '@testing-library/react/cleanup-after-each'
import Card from './index'

describe('<Card>', () => {
    it('renders the title', () => {
        const title = 'hello there i am a test'
        const { getByText } = render(<Card title={title} />)
        const result = getByText(/hello there i am a test/)
        expect(result).toBeTruthy()
    })
})
```

### What you did:

Ran the above test with jest's experimental `--detectLeaks` flag as my tests were crashing my circleci builds. I only have 33 test suites so I figured why not try and see if any have leaks. Going through the tests one by one, the one above was flagged:

```sh
 FAIL  src/atoms/Card/test.tsx
  ● Test suite failed to run

    EXPERIMENTAL FEATURE!
    Your test suite is leaking memory. Please ensure all references are cleaned.

    There is a number of things that can leak memory:
      - Async operations that have not finished (e.g. fs.readFile).
      - Timers not properly mocked (e.g. setInterval, setTimeout).
      - Keeping references to the global scope.

      at node_modules/@jest/core/build/TestScheduler.js:257:24
      at asyncGeneratorStep (node_modules/@jest/core/build/TestScheduler.js:131:24)
      at _next (node_modules/@jest/core/build/TestScheduler.js:151:9)
      at node_modules/@jest/core/build/TestScheduler.js:156:7
      at node_modules/@jest/core/build/TestScheduler.js:148:12
      at onResult (node_modules/@jest/core/build/TestScheduler.js:271:25)
```

### Suggested solution:

I could solve the warning by adding `delete global.result` at the end of the test.. although it's a hack.

```js
    it('renders text correctly', () => {
        const text = 'hello there i am a test'
        const { getByText } = customRender(<Card>{text}</Card>)
        const result = getByText(/hello there i am a test/)
        expect(result).toBeTruthy()
        // here is my "fix":
        delete global.result
    })
```

### Extra notes

I also wanted to say, since this is an experimental feature I am not sure if it's actually a bug or a false alarm! Still, I wanted to raise the issue.

**Comment by kentcdodds** [comment]
Thank you for this! What is that global.result value and what is setting it? 🤔

**Comment by bitttttten** [comment]
The `global.result` was what ended up suppressing the warnings or "fixing" the memory leak. I tried a few different things.

To clarify, it could have been:

```js
        global.result = getByText(/hello there i am a test/)
        expect(global.result).toBeTruthy()
        // here is my "fix":
        delete global.result
```

**Comment by alexkrolick** [comment]
Trying to figure out what's going on here...

Does that mean you were assigning to an undeclared value before? What do you mean "it could have been"?

**Comment by bitttttten** [comment]
"It could have been" meaning "this is also a variantion I could have done to produce the same warning". The difference being `result` was set to the global scope. Sorry if I was not being clear!

Happy to fork and reproduce, I will get back to you when I've done that.

I think a TL;DR is that "on my machine" using getByText was triggering a memory leak from jest's experimental detectLeaks flag. That's the core of the issue! Perhaps it's a false warning.

Then the second part to the issue was, I was curious and I started fiddling around and managed to supress the warning. Perhaps there are other ways to trigger the warning, and perhaps the way I did it was unrelated. I didn't dig too much into that since the main point of the issue was pointing out the warning.. :)
